flink-runtime/src/main/java/org/apache/flink/runtime/concurrent/FutureUtils.java

```diff
@@ -223,6 +224,81 @@
 		}
 	}
 
+	/**
+	 * Retry the given operation with the given delay in between successful completions where the
+	 * result does not match a given predicate.
+	 *
+	 * @param operation to retry
+	 * @param retryDelay delay between retries
+	 * @param deadline A deadline that specifies at what point we should stop retrying
+	 * @param acceptancePredicate Predicate to test whether the result is acceptable
+	 * @param scheduledExecutor executor to be used for the retry operation
+	 * @param <T> type of the result
+	 * @return Future which retries the given operation a given amount of times and delays the retry
+	 *   in case the predicate isn't matched
+	 */
+	public static <T> CompletableFuture<T> retrySuccesfulWithDelay(
+		final Supplier<CompletableFuture<T>> operation,
+		final Time retryDelay,
```

Comment:
Deadline uses `Duration`, this method uses `Time`.

Reply:
The reason is that the other methods in this class use `Time`, but I can switch the new ones to `Duration`.